SharedPackages/VPN/Sources/VPN/WireGuardKit/DNSResolver.swift

```diff
@@ -4,6 +4,18 @@
 import Network
 import Foundation
 
+public protocol DNSResolving {
+    func resolveSync(endpoints: [Endpoint?]) -> [Result<Endpoint, DNSResolutionError>?]
+}
+
+public struct DefaultDNSResolver: DNSResolving {
+    public init() {}
+
+    public func resolveSync(endpoints: [Endpoint?]) -> [Result<Endpoint, DNSResolutionError>?] {
+        DNSResolver.resolveSync(endpoints: endpoints)
```

Comment:
In the interest of getting to testability faster I haven't refactored `DNSResolver` to be dependency injectable on its own, but would be happy to do so.